### PR TITLE
fix: [CDS-39248]: fixing empty script

### PR DESCRIPTION
**Pull request by lasdox**
## Describe your changes

For an artifact stream that has custom repository, if customer has ARTIFACT_COLLECTION_CONFIGURABLE ff enabled, we allow it to have empty shell script.
We should also allow shell scripts that are comment or space only (noop)
This PR fixes this.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39902)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import static io.harness.data.structure.EmptyPredicate.isEmpty;
+
+import io.harness.data.structure.EmptyPredicate;
+
+import com.google.inject.Singleton;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@Singleton
+public class ShellScriptUtils {
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java:12]: The methods in the class can be static?

File: 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  private static final String COMMENT_OPERATOR = "#";
+
+  public boolean isNoopScript(String script) {
+    return isEmpty(script) || isScriptCommentOrSpaceOnly(script);
+  }
+
+  private boolean isScriptCommentOrSpaceOnly(String script) {
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java:19]: Should we introduce this in NG as well?

File: 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java (hunk 1 above)
Comment (lasdox) [reply 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java:12]: Yes we can, i messed it up here, fixing it

File: 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java (hunk 2 above)
Comment (lasdox) [reply 400-rest/src/main/java/software/wings/service/impl/ShellScriptUtils.java:19]: I dont know, do we have similar behaviour in NG? Like custom repositories that only executes if shell script is empty?

**Review by srinivaswings — APPROVED** [review]
> Reviewed 1 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39902)* status: 2. unresolved discussions.

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39902)* status: :shipit: complete! ✓

**Review by inderschera — APPROVED** [review]
> Reviewed 1 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39902)* status: :shipit: complete! ✓

**Pull request merged**